types/postman-collection/index.d.ts

```diff
@@ -478,11 +493,32 @@ export class Request extends Property<RequestDefinition> implements RequestDefin
 }
 
 export interface RequestAuthDefinition extends PropertyDefinition {
-    type?: string;
+    type?:
+        | 'oauth2'
+        | 'hawk'
+        | 'noauth'
+        | 'basic'
+        | 'oauth1'
+        | 'apikey'
+        | 'digest'
+        | 'bearer'
+        | 'awsv4'
+        | 'edgegrid'
+        | 'ntlm';
+    oauth2?: VariableDefinition[];
+    hawk?: VariableDefinition[];
+    basic?: VariableDefinition[];
+    oauth1?: VariableDefinition[];
+    apikey?: VariableDefinition[];
+    digest?: VariableDefinition[];
+    bearer?: VariableDefinition[];
+    awsv4?: VariableDefinition[];
+    edgegrid?: VariableDefinition[];
+    ntlm?: VariableDefinition[];
```

Comment:
I would say here we can probably add all the discriminated union so that we can get to more precise typings when using this:

```ts
export interface oauth2AuthDefinition extends RequestAuthDefinition {
  type: 'oauth2',
  oauth2: VariableDefinition[];
}
```

and so on and so forth for all the auth definitions.

Reply:
I should dig into that more. I resigned when I saw that it is possible to have a `RequestAuthDefinition` shaped like that:
```
{
type: 'basic',
basic: ...,
oauth2: ...,
}
```

However, I'll take a deeper look into that. The approach you mentioned is way better 👍

Reply:
@karol-maciaszek Did you get any change tf checking this out?

Reply:
Ok, I'm quite unsure whether I should enforce the type so strictly.
Personally I would, but it won't be compatible with at lest this example from Postman Collection SDK tests:
https://github.com/postmanlabs/postman-collection/blob/develop/examples/collection-v2.json#L301

Reply:
I'm quite unsure whether I should enforce the type so strictly.
Personally I would, but it won't be compatible with at least this example from Postman Collection SDK tests:
https://github.com/postmanlabs/postman-collection/blob/develop/examples/collection-v2.json#L301

Reply:
Agreed, we'll keep it this way for now, thanks.